src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -19,22 +19,24 @@ import scala.concurrent.{ExecutionContext, Future}
 /**
   * Created by davidan on 10/7/16.
   */
-trait RestJsonClient extends FireCloudRequestBuilding {
+trait RestJsonClient extends FireCloudRequestBuilding with PerformanceLogging {
   implicit val system: ActorSystem
   implicit val executionContext: ExecutionContext
 
-  def unAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None): Future[HttpResponse] = {
+  def unAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None): Future[HttpResponse] = {
     implicit val userInfo:WithAccessToken = null
-    doRequest(None)(req, compressed, useFireCloudHeader, connector)
+    doRequest(None)(req, compressed, useFireCloudHeader, connector, label)
   }
 
-  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None)(implicit userInfo: WithAccessToken): Future[HttpResponse] =
-    doRequest(Option(addCredentials(userInfo.accessToken)))(req, compressed, useFireCloudHeader, connector)
+  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None)(implicit userInfo: WithAccessToken): Future[HttpResponse] =
+    doRequest(Option(addCredentials(userInfo.accessToken)))(req, compressed, useFireCloudHeader, connector, label)
 
-  def adminAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None): Future[HttpResponse] =
-    doRequest(Option(addAdminCredentials))(req, compressed, useFireCloudHeader, connector)
+  def adminAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None): Future[HttpResponse] =
+    doRequest(Option(addAdminCredentials))(req, compressed, useFireCloudHeader, connector, label)
+
+  private def doRequest(addCreds: Option[RequestTransformer])(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None): Future[HttpResponse] = {
+    val tick = if (label.nonEmpty) System.currentTimeMillis() else -1
```

Comment:
I'd move the definition of `tick` closer to where it's used for locality and avoid the magic number `-1` (e.g. `val NoLabel = -1` and use `NoLabel` downstream) for readability.

Reply:
done